### ISSUE TITLE
What are the hardware requirements for launching PrairieLearn locally

### ISSUE THREAD

**Issue by jakekugel** [PrairieLearn/PrairieLearn#7927]
Hello,

I'm not sure if this is the best place to raise this question, please let me know if I should ask it somewhere else.  I'm new to PrairieLearn and I am trying to launch the application on my laptop for local course development using instructions [here](https://prairielearn.readthedocs.io/en/latest/installing/).  I am seeing some pretty slow performance though, and I am wondering if my laptop meets the minimum system requirements.  I am launching the application using:

`docker run -it --rm -p 3000:3000 -v C:\Users\jake\prairielearn-test-course:/course prairielearn/prairielearn`

Here are the messages I see, and how much time elapsed before each:

`Starting PrairieLearn` - 13 seconds
`info: Workspace server listening on port 8081` - 1 minute 10 seconds
`error: Error creating Python worker timeout exceeded, killing CodeCallerNative child` - 3 minutes 41 seconds
`info: PrairieLearn server ready, press Control-C to quit` - 3 minutes 41 seconds

I have a 2019 HP Pavilion laptop with an Intel Core i7 and 12 GB of RAM, do you think my hardware is sufficient?

Thanks,
Jake

**Comment by jonatanschroeder** [comment]
I don't have a direct answer to your question, but given you're working with Windows I'm wondering if you'd be able to test your environment [using WSL2](https://prairielearn.readthedocs.io/en/latest/installing/#running-prairielearn-from-a-wsl2-instance). From personal experience it tends to be a bit faster. You will need to make sure [WSL2 is installed](https://learn.microsoft.com/en-us/windows/wsl/install).

**Comment by eliotwrobson** [comment]
I don't think that we have any tested system requirements for windows. However, I have tried running PL on my desktop, and it was also very slow. Unfortunately docker can be pretty taxing on your system on Windows, so that might be the culprit.

I would highly recommend dual-booting Linux for doing development (though I know that can be pretty involved).

**Comment by nwalters512** [comment]
I'll second WSL2; specifically, make sure you're launching the docker container _within_ WSL2. To the extent that people have a good experience running Docker in Windows, I think that's usually how they do it.

I'm going to convert this into a discussion. If it turns out that there's anything we have to change or document, we can open an issue for that specifically!
